internal/locate/region_request.go

```diff
@@ -604,7 +605,7 @@ func (state *accessFollower) isCandidate(idx AccessIndex, replica *replica) bool
 		// The request can only be sent to the leader.
 		((state.option.leaderOnly && idx == state.leaderIdx) ||
 			// Choose a replica with matched labels.
-			(!state.option.leaderOnly && (state.tryLeader || idx != state.leaderIdx) && replica.store.IsLabelsMatch(state.option.labels))) &&
+			(!state.option.leaderOnly && (state.tryLeader || idx != state.leaderIdx) && (state.lastIdx >= 0 || replica.store.IsLabelsMatch(state.option.labels)))) &&
```

Comment:
Why adding `(state.lastIdx >= 0 ||` here?

Reply:
I change this condition to `leaderExhaust`.

The try order is local follower -> leader -> remote follower.

When `leaderExhaust` is true, we need to ignore the labels to select the remote follower.

Reply:
Is is working as expected when the `local follower` is the leader peer picked first and it's unavailable for both stale and follower read requests?
It's better to comment about the details here.